packages/markdown-docx/src/ToCiceroMarkVisitor.js

```diff
@@ -32,8 +32,11 @@ class ToCiceroMarkVisitor {
         // All the nodes generated from given OOXML
         this.nodes = [];
 
-        // contains the realtionship part of a given OOXML
+        // contains the relationship part of a given OOXML
         this.relationshipXML = [];
+
+        // contains the nodes present in a conditional or optional OOXML
+        this.conditionalOrOptionalNodes = [];
```

Comment:
Same logic as below. I think it would be easier to implement it here because you actually return the nodes here. So you can recursively create an entire node for Optional node without having this array as a side-effect.

```
constructCiceroNodes() {
   if OPTIONAL { 
      ooxmlToBeDisplayed = constructCiceroNodes(whenSome) // ofcouse this depends upon `hasSome` so you would need a check for it.
      ooxmlToBeHidden = constructCiceroNodes(whenNode)
  }
}
```